plugin/documents.py

```diff
@@ -419,9 +423,16 @@ def _clear_highlight_regions(self) -> None:
         for kind in global_settings.document_highlight_scopes.keys():
             self.view.erase_regions("lsp_highlight_{}".format(kind))
 
+    def is_in_higlighted_region(self, point: int) -> bool:
+        for kind in global_settings.document_highlight_scopes.keys():
+            regions = self.view.get_regions("lsp_highlight_{}".format(kind))
+            for r in regions:
+                if r.contains(point):
+                    return True
+        return False
+
     def _do_highlights(self) -> None:
-        self._clear_highlight_regions()
-        if len(self.view.sel()) != 1:
+        if not len(self.view.sel()):
```

Comment:
I don't think we should allow multiple cursors unless we really support that by highlighting all different words under all cursors.

Reply:
Here is how VS Code handles it: 
![image](https://user-images.githubusercontent.com/22029477/89731605-ac36e280-da48-11ea-8240-91398b110fa2.png)

We had the same issue with the completion request, because we can trigger AC, but the cursors can be on different places (symbols).
In that case the request is only send to the primary cursor(in our case that is the view.sel()[0].b point)

So that is the reason why I think this logic should stay.
If you still believe that we should handle this differently, I will revert the commit. :)

Reply:
Completions are a bit different case. With multiple cursors the completion from the "primary" cursor is inserted at all cursor positions. So that is fine.

Here you have multiple cursors but only the primary one does trigger selection while other cursors don't have any function so I think it might be a bit confusing in the current state. I think it would be better to not select anything when there are multiple cursors. Like before

Reply:
But feel free to leave as is. It's not that important.

Reply:
I would leave it just because of this:
When someone has multi cursors and edits the selections, I want to allow the highlight request. 

![output](https://user-images.githubusercontent.com/22029477/89778204-6728b400-db0d-11ea-9456-ceb90ddc7e88.gif)

If I were to leave the highlight request to be send only where there is only one selection, in that case we would not see the highlights. Here is an example for `if len(self.view.sel()) != 1:`:
![output](https://user-images.githubusercontent.com/22029477/89778675-38f7a400-db0e-11ea-809d-3c363f4c9dd5.gif)